From a4c77cf4f85e3b60b660338bed2b2d371a193ec6 Mon Sep 17 00:00:00 2001
From: Martin Jansa <martin.jansa@gmail.com>
Date: Tue, 27 Feb 2024 12:57:08 +0100
Subject: [PATCH] Revert "Disable GPU acceleration on VMware on Linux"

This reverts commit 6265ef99615f9e1dfe838f3684a751eeeb91d492.

Works for us in VirtualBox and with this webapp-mgr fails with:
Feb 23 10:32:57 qemux86-64 systemd[1]: Starting meta-webos - "webapp-mgr.service"...
Feb 23 10:32:57 qemux86-64 systemd[1]: Started meta-webos - "webapp-mgr.service".
Feb 23 10:32:57 qemux86-64 webapp-mgr.sh[334]: [334:334:INFO:ozone_platform.cc(90)] Ozone platform name: 'wayland'
Feb 23 10:32:57 qemux86-64 webapp-mgr.sh[334]: [334:668:ERROR:egl_util.cc(44)] Failed to load GLES library: /usr/lib/libGLESv2.so: /usr/lib/libGLESv2.so: cannot open shared object file: No such file or directory
...
Feb 23 10:32:58 qemux86-64 webapp-mgr.sh[334]: [334:668:ERROR:gpu_channel_manager.cc(884)] Failed to create GLES3 context, fallback to GLES2.
Feb 23 10:32:58 qemux86-64 webapp-mgr.sh[334]: [334:668:ERROR:gpu_channel_manager.cc(895)] ContextResult::kFatalFailure: Failed to create shared context for virtualization.
Feb 23 10:32:58 qemux86-64 webapp-mgr.sh[334]: [334:668:ERROR:shared_image_stub.cc(451)] SharedImageStub: unable to create context
Feb 23 10:32:58 qemux86-64 webapp-mgr.sh[334]: [334:668:ERROR:gpu_channel.cc(601)] GpuChannel: Failed to create SharedImageStub
Feb 23 10:32:58 qemux86-64 webapp-mgr.sh[334]: [334:668:ERROR:gpu_channel_manager.cc(884)] Failed to create GLES3 context, fallback to GLES2.
Feb 23 10:32:58 qemux86-64 webapp-mgr.sh[334]: [334:668:ERROR:gpu_channel_manager.cc(895)] ContextResult::kFatalFailure: Failed to create shared context for virtualization.
Feb 23 10:32:58 qemux86-64 webapp-mgr.sh[334]: [334:668:ERROR:shared_image_stub.cc(451)] SharedImageStub: unable to create context
Feb 23 10:32:58 qemux86-64 webapp-mgr.sh[334]: [334:668:ERROR:gpu_channel.cc(601)] GpuChannel: Failed to create SharedImageStub

Signed-off-by: Martin Jansa <martin.jansa@gmail.com>
---
Upstream-Status: Backport [submissions/23]

 src/gpu/config/software_rendering_list.json | 12 ------------
 1 file changed, 12 deletions(-)

diff --git a/src/gpu/config/software_rendering_list.json b/src/gpu/config/software_rendering_list.json
index 20ea3b557d..99a06ce311 100644
--- a/src/gpu/config/software_rendering_list.json
+++ b/src/gpu/config/software_rendering_list.json
@@ -1665,18 +1665,6 @@
         "accelerated_video_encode"
       ]
     },
-    {
-      "id": 176,
-      "description": "VMware is buggy on Linux",
-      "cr_bugs": [1327939],
-      "os": {
-        "type": "linux"
-      },
-      "vendor_id": "0x15ad",
-      "features": [
-        "all"
-      ]
-    },
     {
       "id": 177,
       "description": "Apple Software Renderer used under VMWare experiences synchronization issues with GPU Raster",
